Pika 0.7.0 fails to compile on MacOS

Compilation of pika in release mode fails in spack.

The full spec of the failing build is:
`^pika@0.7.0%clang@12.0.1~apex~cuda~examples+generic_coroutines~ipo+mpi~rocm~tracy build_type=Release cxxstd=17 malloc=tcmalloc arch=darwin-catalina-skylake`

Note: in debug mode it works. 
`^pika@0.7.0%clang@12.0.1~apex~cuda~examples+generic_coroutines~ipo+mpi~rocm~tracy build_type=Debug cxxstd=17 dev_path=/Users/rasolca/code/spack_debug/pika-0.7.0 malloc=tcmalloc arch=darwin-catalina-skylake`

Error messsage:
```
  >> 263    Undefined symbols for architecture x86_64:
     264      "void pika::debug::detail::print_dec<unsigned long>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, unsigned long const&, int)", referenced from:
     265          pika::mpi::experimental::detail::operator<<(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, pika::mpi::experimental::detail::mpi_data const&) in unity_0_cxx.cxx.o
     266    ld: symbol(s) not found for architecture x86_64
```

--- a/libs/pika/debugging/src/print.cpp
+++ b/libs/pika/debugging/src/print.cpp
@@ -62,6 +62,11 @@ namespace NS_DEBUG {
         std::ostream&, std::int64_t const&, int);
     template PIKA_EXPORT void print_dec(
         std::ostream&, std::uint64_t const&, int);
+#if defined(__APPLE__)
+    // Explicit instantiation necessary to solve undefined symbol for MacOS
+    template PIKA_EXPORT void print_dec(
+        std::ostream&, unsigned long const&, int);
+#endif
 
     template PIKA_EXPORT void print_dec(
         std::ostream&, std::atomic<int> const&, int);

